instrumentation/kubernetes-client-7.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/kubernetesclient/KubernetesClientTracer.java

```diff
@@ -73,24 +59,42 @@ protected String requestHeader(Request request, String name) {
   }
 
   @Override
-  protected String responseHeader(Response response, String name) {
-    return response.header(name);
+  protected String responseHeader(ApiResponse<?> response, String name) {
+    Map<String, List<String>> responseHeaders =
+        response.getHeaders() == null ? Collections.emptyMap() : response.getHeaders();
+    return responseHeaders.getOrDefault(name, Collections.emptyList()).stream()
+        .findFirst()
+        .orElse(null);
```

Comment:
this is not a very convenient api 🤷‍♂️

Reply:
I'd do `return !values.isEmpty() ? values.get(0) : null` instead of streaming

Reply:
Null safety in Java is fun in general 😂